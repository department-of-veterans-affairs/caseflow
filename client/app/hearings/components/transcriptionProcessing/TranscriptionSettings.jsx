/* eslint-disable no-nested-ternary */
/* eslint-disable max-len */
import React from 'react';
import { css } from 'glamor';
import PropTypes from 'prop-types';
import AppSegment from '@department-of-veterans-affairs/caseflow-frontend-toolkit/components/AppSegment';
import Button from 'app/components/Button';
import Link from '@department-of-veterans-affairs/caseflow-frontend-toolkit/components/Link';
import ToggleSwitch from '../../../components/ToggleSwitch/ToggleSwitch';
import { PencilIcon } from '../../../components/icons/PencilIcon';
import Alert from '../../../components/Alert';
import ApiUtil from '../../../util/ApiUtil';
import COPY from '../../../../COPY';
import { RemoveContractorModal } from './RemoveContractorModal';

const buttonStyle = css({
  padding: '1rem 2.5rem 2rem 0',
  display: 'inline-block',
});

const contractorButtonStyle = css({
  paddingLeft: '41.55rem',
});

const headerContainerStyling = css({
  margin: '1.5rem 0 3rem 0',
  padding: '0',
  '& > *': {
    display: 'inline-block',
    paddingRight: '15px',
    // paddingLeft: '15px',
    verticalAlign: 'middle',
    margin: 0,
  },
});

const headerStyling = css({
  paddingLeft: 0,
});

const instructionListStyle = css({
  listStyle: 'none',
  margin: '0',
  padding: '1.5rem 0 2rem 0',
  fontSize: '19px',
});

const returnLinkStyle = css({
  padding: '1.5rem 0 2rem 0rem',
});

const toggleStyle = css({
  padding: '1.5rem 0 2rem 25rem',
});

const userListItemStyle = css({
  display: 'flex',
  flexWrap: 'wrap',
  borderTop: '.1rem solid #d6d7d9',
  padding: '4rem 0 2rem',
  margin: '0',
});

const EditContractorLink = () => (
  <Button linkStyling>
    <span {...css({ marginRight: '1px', marginLeft: '5px' })}>
      Edit Information
    </span>
    <span {...css({ position: 'absolute' })}>
      <PencilIcon size={25} />
    </span>
  </Button>
);

const EditHearingsSentLink = () => (
  <Button linkStyling>
    <span {...css({ marginRight: '1px', marginLeft: '5px' })}>Edit Total</span>
    <span {...css({ position: 'absolute' })}>
      <PencilIcon size={25} />
    </span>
  </Button>
);

export default class TranscriptionSettings extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      isRemoveModalOpen: false,
      loading: true,
      alert: {
        title: '',
        message: '',
        type: '',
      },
      isAddEditOpen: false,

      contractors: [],
    };
  }

  toggleRemoveModal = () => {
    this.setState((prevState) => ({
      isRemoveModalOpen: !prevState.isRemoveModalOpen,
    }));
  };

  handleRemoveContractor = (contractorId) => {
    return new Promise((resolve, reject) => {
      ApiUtil.delete(`/hearings/find_by_contractor/${contractorId}`).
        then(() => {
          this.setState(
            {
              isRemoveModalOpen: false,
            },
            () => {
              this.getContractors();
              this.confirmRemoveModal({
                title: 'Success',
                message: COPY.TRANSCRIPTION_SETTINGS_CONTRACTOR_REMOVAL_SUCCESS,
                type: 'success',
              });
              resolve();
            }
          );
        }).
        catch((error) => {
          console.error(error);
          this.setState(
            {
              alert: {
                title: 'Error',
                message: COPY.TRANSCRIPTION_SETTINGS_CONTRACTOR_REMOVAL_FAIL,
                type: 'error',
              },
              isRemoveModalOpen: false,
            },
            () => {
              reject(error);
            }
          );
        });
    });
  };

  confirmRemoveModal = (alert) => {
    this.setState({ alert });
    this.toggleRemoveModal();
  };

  getContractors = () => {
    ApiUtil.get('/hearings/find_by_contractor').then((response) => {
      this.setState({
        contractors: response.body.transcription_contractors,
        loading: false,
      });
    });
  };

  componentDidMount() {
    this.getContractors();
  }

  addContractorButton = () => (
    <div {...buttonStyle}>
      <Button
        name={COPY.TRANSCRIPTION_SETTINGS_ADD}
        id="Add-contractor"
        classNames={['usa-button-primary']}
        // on click add contractor modal opens
      />
    </div>
  );

  removeContractorButton = () => (
    <div {...buttonStyle}>
      <Button
        name={COPY.TRANSCRIPTION_SETTINGS_REMOVE}
        id="Remove-contractor"
        classNames={['usa-button-secondary']}
        onClick={() => this.toggleRemoveModal()}
        // on click contractor is removed
      />
    </div>
  );

  mainContent = () => {
    const listOfContractors = () => {
      // pass in and iterate over contractors
      return (
        <React.Fragment>
          <div {...userListItemStyle}>
            <div>
              <ul {...instructionListStyle}>
                <h2>
                  {COPY.TRANSCRIPTION_SETTINGS_CONTRACTOR_NAME}
                  <EditContractorLink />
                </h2>
                <li>
                  <strong>{COPY.TRANSCRIPTION_SETTINGS_BOX_LINK}</strong>
                  `https://box.com/`
                </li>
                <li>
                  <strong>{COPY.TRANSCRIPTION_SETTINGS_POC_ADDRESS}</strong>
                  `Address here`
                </li>
                <span>
                  <li>
                    <strong>{COPY.TRANSCRIPTION_SETTINGS_HEARINGS_SENT}</strong>
                    `50 of 160`
                    <EditHearingsSentLink />
                  </li>
                </span>
              </ul>
            </div>
            <span {...toggleStyle}>
              <h3>{COPY.TRANSCRIPTION_SETTINGS_WORK_TOGGLE}</h3>
              <ToggleSwitch />
            </span>
          </div>
        </React.Fragment>
      );
    };

    return (
      <React.Fragment>
        <div>
          <h1 className="cf-margin-bottom-0" {...headerStyling}>
            {COPY.TRANSCRIPTION_SETTINGS_HEADER}
          </h1>
          <div {...headerContainerStyling}>
            <h2 {...headerStyling}>{COPY.TRANSCRIPTION_SETTINGS_SUBHEADER}</h2>
            <span {...contractorButtonStyle}>
              {this.removeContractorButton()}
              {this.addContractorButton()}
            </span>
          </div>
        </div>
        <div>{listOfContractors()}</div>
      </React.Fragment>
    );
  };

  render = () => (
    <>
      {this.state.alert.title && (
        <Alert
          title={this.state.alert.title}
          message={this.state.alert.message}
          type={this.state.alert.type}
        />
      )}
      <AppSegment filledBackground>
        <div {...returnLinkStyle}>
          <span>
            <Link linkStyling>&lt; {COPY.TRANSCRIPTION_QUEUE_LINK}</Link>
            &nbsp;
          </span>
        </div>
        <div>{this.mainContent()}</div>
        {this.state.isRemoveModalOpen && (
          <RemoveContractorModal
            onCancel={this.toggleRemoveModal}
            onConfirm={this.handleRemoveContractor}
            contractors={this.state.contractors}
            title={COPY.TRANSCRIPTION_SETTINGS_REMOVE_CONTRACTOR_MODAL_TITLE}
          />
        )}
      </AppSegment>
    </>
  );
}

TranscriptionSettings.propTypes = {
  contractors: PropTypes.array.isRequired,
};
