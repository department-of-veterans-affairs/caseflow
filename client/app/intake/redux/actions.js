import { ACTIONS } from '../constants';
import ApiUtil from '../../util/ApiUtil';
import { formatDateStringForApi } from '../../util/DateUtil';

export const startNewIntake = () => ({
  type: ACTIONS.START_NEW_INTAKE
});

export const setFileNumberSearch = (fileNumber) => ({
  type: ACTIONS.SET_FILE_NUMBER_SEARCH,
  payload: {
    fileNumber
  }
});

export const doFileNumberSearch = (fileNumberSearch) => (dispatch) => {
  dispatch({
    type: ACTIONS.FILE_NUMBER_SEARCH_START
  });

  return ApiUtil.post('/intake', { data: { file_number: fileNumberSearch } }).
    then(
      (response) => {
        const responseObject = JSON.parse(response.text);

        dispatch({
          type: ACTIONS.FILE_NUMBER_SEARCH_SUCCEED,
          payload: {
            intakeId: responseObject.id,
            name: responseObject.veteran_name,
            formName: responseObject.veteran_form_name,
            fileNumber: responseObject.veteran_file_number
          }
        });
      },
      (error) => {
        const responseObject = JSON.parse(error.response.text);

        dispatch({
          type: ACTIONS.FILE_NUMBER_SEARCH_FAIL,
          payload: {
            errorCode: responseObject.error_code
          }
        });
      }
    );
};

export const setSelectedOption = (optionSelected) => ({
  type: ACTIONS.SET_SELECTED_OPTION,
  payload: {
    optionSelected
  }
});

export const setReceiptDate = (receiptDate) => ({
  type: ACTIONS.SET_RECEIPT_DATE,
  payload: {
    receiptDate
  }
});

export const submitReview = (rampElection) => (dispatch) => {
  dispatch({
    type: ACTIONS.SUBMIT_REVIEW_START
  });

  const data = {
    option_selected: rampElection.optionSelected,
    receipt_date: formatDateStringForApi(rampElection.receiptDate)
  };

  return ApiUtil.patch(`/intake/ramp/${rampElection.intakeId}`, { data }).
    then(
      () => dispatch({ type: ACTIONS.SUBMIT_REVIEW_SUCCEED }),
      () => dispatch({ type: ACTIONS.SUBMIT_REVIEW_FAIL })
    );
};
