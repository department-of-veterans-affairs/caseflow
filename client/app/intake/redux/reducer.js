import { ACTIONS, REQUEST_STATE } from '../constants';
import { update } from '../../util/ReducerUtil';

export const mapDataToInitialState = (data = {}) => ({
  veteran: {
    name: null,
    formName: null,
    fileNumber: null
  },
  inputs: {
    fileNumberSearch: '',
    receiptDateStr: '',
    veteranResponse: null
  },
  requestStatus: {
    fileNumberSearch: REQUEST_STATE.NOT_STARTED,
    submitReview: REQUEST_STATE.NOT_STARTED
  },
  rampElection: {
    intakeId: data.intakeId,
    optionSelected: null,
    receiptDate: null
  },
  searchError: null
});

// The keys in this object need to be snake_case
// because they're being matched to server response values.
const searchErrors = {
  invalid_file_number: {
    title: 'Veteran ID not found',
    body: 'Please enter a valid Veteran ID and try again.'
  },
  veteran_not_found: {
    title: 'Veteran ID not found',
    body: 'Please enter a valid Veteran ID and try again.'
  },
  veteran_not_accessible: {
    title: 'You don\'t have permission to view this veteran\'s informationâ€‹',
    body: 'Please enter a valid Veteran ID and try again.'
  },
  did_not_receive_ramp_election: {
    title: 'No opt-in letter was sent to this veteran',
    body: "An opt-in letter was not sent to this Veteran, so this form can't be processed" +
      'Please enter a valid Veteran ID below.'
  },
  default: {
    title: 'Something went wrong',
    body: 'Please try again. If the problem persists, please contact Caseflow support.'
  }
};


export const reducer = (state = mapDataToInitialState(), action) => {
  switch (action.type) {
  case ACTIONS.START_NEW_INTAKE:
    return mapDataToInitialState();
  case ACTIONS.SET_FILE_NUMBER_SEARCH:
    return update(state, {
      inputs: {
        fileNumberSearch: {
          $set: action.payload.fileNumber
        }
      }
    });
  case ACTIONS.SET_SELECTED_OPTION:
    return update(state, {
      rampElection: {
        optionSelected: {
          $set: action.payload.optionSelected
        }
      }
    });
  case ACTIONS.SET_RECEIPT_DATE:
    return update(state, {
      rampElection: {
        receiptDate: {
          $set: action.payload.receiptDate
        }
      }
    });
  case ACTIONS.FILE_NUMBER_SEARCH_START:
    return update(state, {
      requestStatus: {
        fileNumberSearch: {
          $set: REQUEST_STATE.IN_PROGRESS
        }
      }
    });
  case ACTIONS.FILE_NUMBER_SEARCH_SUCCEED:
    return update(state, {
      requestStatus: {
        fileNumberSearch: {
          $set: REQUEST_STATE.SUCCEEDED
        }
      },
      veteran: {
        name: {
          $set: action.payload.name
        },
        formName: {
          $set: action.payload.formName
        },
        fileNumber: {
          $set: action.payload.fileNumber
        }
      },
      rampElection: {
        intakeId: {
          $set: action.payload.intakeId
        }
      }
    });
  case ACTIONS.FILE_NUMBER_SEARCH_FAIL:
    return update(state, {
      searchError: {
        $set: searchErrors[action.payload.errorCode] || searchErrors.default
      },
      requestStatus: {
        fileNumberSearch: {
          $set: REQUEST_STATE.FAILED
        }
      }
    });
  case ACTIONS.SUBMIT_REVIEW_START:
    return update(state, {
      requestStatus: {
        submitReview: {
          $set: REQUEST_STATE.IN_PROGRESS
        }
      }
    });
  case ACTIONS.SUBMIT_REVIEW_SUCCEED:
    return update(state, {
      requestStatus: {
        submitReview: {
          $set: REQUEST_STATE.SUCCEEDED
        }
      }
    });
  case ACTIONS.SUBMIT_REVIEW_FAIL:
    return update(state, {
      requestStatus: {
        submitReview: {
          $set: REQUEST_STATE.FAILED
        }
      }
    });
  default:
    return state;
  }
};
