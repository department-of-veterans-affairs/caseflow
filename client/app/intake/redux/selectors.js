import { createSelector } from 'reselect';
import { RAMP_INTAKE_STATES } from '../constants';

const getRampElection = (state) => state.rampElection;

export const getRampElectionStatus = createSelector(
  [getRampElection],
  (rampElection) => {
    if (rampElection.isComplete) {
      return RAMP_INTAKE_STATES.COMPLETED;
    } else if (rampElection.isReviewed) {
      return RAMP_INTAKE_STATES.REVIEWED;
    } else if (rampElection.intakeId) {
      return RAMP_INTAKE_STATES.STARTED;
    }

    return RAMP_INTAKE_STATES.NONE;
  }
);
