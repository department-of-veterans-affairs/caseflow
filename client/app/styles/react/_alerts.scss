
.usa-alert {

  &.fixed {
    position: fixed;
    top: 0;
    margin-top: 0;
    z-index: 999;
    left: 50%;
    transform: translate(-50%, 0);
  }

  // Remove after USWDS V.2.0
  &.usa-alert-slim {
    background-size: 4rem;

    .usa-alert-body {
      padding-left: 3.5rem;
    }

    .usa-alert-text {
      font-size: 2rem;
    }
  }

  .usa-alert-text {
    line-height: 1.5;
  }
}
