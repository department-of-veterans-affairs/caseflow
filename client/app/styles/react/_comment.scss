.comment-control-button-container {
  display: flex;
  width: 100%;
  padding: 3px 0;
  justify-content: space-between;
}

.comment-save-button-container {
  display: inline-block;
  width: 100%;
  margin-top: 5px;
  margin-bottom: 17px;
}

.comment-control-button {
  padding: 0;

  &.horizontal {
    text-align: left;
  }
}

.comment-control-button-divider {
  padding: 0 10px;
}

.comment-container-selected {
  box-shadow: 3px 3px 5px $color-gray-light;
}

.comment-container {
  word-wrap: break-word;
  word-break: break-word;
  background-color: $color-white;
  padding: 15px;
  cursor: pointer;

  &:not(.comment-horizontal-container) {
    margin-bottom: 15px;
    border-radius: 3px;
    border: 1px solid $color-gray-lighter;
  }

  &.comment-horizontal-container {
    padding: 25px 25px 25px 40px;
  }
}

// Max width is enforced so the size of this comment box
// fits with existing textareas in style guide
.comment-size-container {
  max-width: 430px;

  .cf-form-textarea {
    margin-bottom: 20px;
  }
}

.comment-horizontal-container {
  display: inline-block;
  width: 100%;
}

.comment-horizontal-spacing {
  margin-right: 49px;
}

.commentIcon-container {
  position: absolute;
  z-index: 10;
  // These dimensions need to match the comment icon SVG dimensions.
  width: 40px;
  height: 40px;
}

