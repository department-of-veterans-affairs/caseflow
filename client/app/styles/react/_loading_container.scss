.loadingContainer-positioning {
  position: absolute;
  width: 100%;
  height: 100%;
}

.loadingContainer-table {
  width: 100%;
  height: 100%;
  display: table;
}

.loadingContainer-table-cell {
  text-align: center;
  display: table-cell;
  vertical-align: middle;
}

.loadingContainer-content {
  position: relative;
  z-index: 1;

  iframe {
    width: 100%;
    height: 80vh;

    &.form9Viewer {
      height: 500px;
    }
  }
}

.loadingContainer-container {
  position: relative;
  overflow: hidden;
}

.cf-loading-button-text {
  display: inline-block;
  margin-right: 10px;
  line-height: 0;
  position: relative;
}

.cf-loading-message {
  .loading-message-text {
    display: inline-block;
    font-style: italic;
    font-weight: bolder;
  }

}

.cf-loading-button-symbol {
  display: inline-block;
}
