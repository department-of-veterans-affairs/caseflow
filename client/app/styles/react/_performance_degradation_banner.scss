.usa-banner {
  background-color: $color-gold-lightest;
  text-align: center;
}

.usa-banner-inner {
  padding: .8em;
}

.banner-icon {
  display: inline-block;
  vertical-align: text-top;
  // height: 25px;
}

.banner-text {
  padding-left: 6px;
}
