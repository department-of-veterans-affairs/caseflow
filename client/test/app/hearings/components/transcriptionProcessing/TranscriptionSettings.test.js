import React from 'react';
import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';
import { MemoryRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { applyMiddleware, createStore, compose } from 'redux';
import thunk from 'redux-thunk';

import TranscriptionSettings from '../../../../../app/hearings/components/transcriptionProcessing/TranscriptionSettings';

const createStoreWithReducer = (initialState) => {
  const reducer = (state = initialState) => state;

  return createStore(reducer, compose(applyMiddleware(thunk)));
};

const renderTranscriptionSettings = () => {
  const store = createStoreWithReducer({ components: {} });

  return render(
    <Provider store={store}>
      <Router>
        < TranscriptionSettings />
      </Router>
    </Provider>
  );
};

it('does render transcription settings information', async () => {
  renderTranscriptionSettings();

  expect(await screen.findByText(/Transcription Settings/)).toBeInTheDocument();
  expect(await screen.findByText(/Edit Current Contractors/)).toBeInTheDocument();

  expect(await screen.findByText(/Link to box.com:/)).toBeInTheDocument();
  expect(await screen.findByText(/POC:/)).toBeInTheDocument();
  expect(await screen.findByText(/Hearings sent to Contractor A this week:/)).toBeInTheDocument();
});
