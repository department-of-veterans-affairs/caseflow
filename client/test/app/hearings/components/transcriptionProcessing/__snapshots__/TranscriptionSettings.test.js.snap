// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does render transcription settings information 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-css-818q9q=""
      >
        <span>
          <a>
            &lt; 
            Back to Transcription queue
          </a>
           
        </span>
      </div>
      
      <section
        class="cf-app-segment cf-app-segment--alt"
      >
        <div>
          <div>
            <h1
              class="cf-margin-bottom-0"
            >
              Transcription settings
            </h1>
            <div
              data-css-89ju4s=""
            >
              <h2>
                Edit current contractors
              </h2>
              <span
                class="cf-push-right"
              >
                <div
                  data-css-19a7m7c=""
                >
                  <span>
                    <button
                      class="usa-button-secondary usa-button"
                      id="Remove-contractor"
                      type="button"
                    >
                      Remove contractor
                    </button>
                  </span>
                </div>
                <div
                  data-css-19a7m7c=""
                >
                  <span>
                    <button
                      class="usa-button-primary usa-button"
                      id="Add-contractor"
                      type="button"
                    >
                      Add contractor
                    </button>
                  </span>
                </div>
              </span>
            </div>
          </div>
          <div>
            <p
              class="no-results-found-styling"
            >
              No contractors found
            </p>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <div
      data-css-818q9q=""
    >
      <span>
        <a>
          &lt; 
          Back to Transcription queue
        </a>
         
      </span>
    </div>
    
    <section
      class="cf-app-segment cf-app-segment--alt"
    >
      <div>
        <div>
          <h1
            class="cf-margin-bottom-0"
          >
            Transcription settings
          </h1>
          <div
            data-css-89ju4s=""
          >
            <h2>
              Edit current contractors
            </h2>
            <span
              class="cf-push-right"
            >
              <div
                data-css-19a7m7c=""
              >
                <span>
                  <button
                    class="usa-button-secondary usa-button"
                    id="Remove-contractor"
                    type="button"
                  >
                    Remove contractor
                  </button>
                </span>
              </div>
              <div
                data-css-19a7m7c=""
              >
                <span>
                  <button
                    class="usa-button-primary usa-button"
                    id="Add-contractor"
                    type="button"
                  >
                    Add contractor
                  </button>
                </span>
              </div>
            </span>
          </div>
        </div>
        <div>
          <p
            class="no-results-found-styling"
          >
            No contractors found
          </p>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
